frontend/src/views/DailyView/DailyView.jsx

```diff
@@ -26,28 +32,32 @@ const DailyDrugs = () => {
         </div>
         <div className="drug-box-container">
           <Box className="drug-box" sx={{ display: 'flex', borderRadius: 15 }}>
-            <Box className="drug-box-compartment 1" sx={{ borderRadius: 10, margin: 3 }}>
-              <Typography align="center" className="time-of-day" variant="h6">
-                {DAYTIMES[0]}
-              </Typography>
-            </Box>
-
-            <Box className="drug-box-compartment 2" sx={{ borderRadius: 10, margin: 3 }}>
-              <Typography align="center" className="time-of-day" variant="h6">
-                {DAYTIMES[1]}
-              </Typography>
-            </Box>
-
-            <Box className="drug-box-compartment 3" sx={{ borderRadius: 10, margin: 3 }}>
-              <Typography align="center" className="time-of-day" variant="h6">
-                {DAYTIMES[2]}
-              </Typography>
-            </Box>
+            {DAYTIMES.map((daytime) => (
+              <Box className="drug-box-compartment" key={daytime} sx={{ borderRadius: 10, margin: 3 }}>
+                <Typography align="center" className="time-of-day" variant="h6">
+                  {daytime}
+                </Typography>
+                {drugList.map((drug) =>
+                  drug.daytime.map(
+                    (dt) =>
+                      dt.toUpperCase() === daytime && (
+                        <Pill
+                          key={`${drug.name}_${drug.expirationDate}`}
+                          typeOfMedication={drug.type}
+                          name={drug.name}
+                          showExpirationDate={false}
+                          expirationDate={drug.expirationDate}
+                        />
+                      ),
+                  ),
+                )}
```

Comment:
I thing it would be much cleaner if drugList was filtered first 
eg. 
drugList.filter(drug => drug.daytime === daytime).map(drug => //each drug here has section's daytime, so we can simply create Pills here)

Reply:
change introduced